compact_str/src/repr/mod.rs

```diff
@@ -810,6 +837,58 @@ fn ensure_read(value: usize) -> usize {
     value
 }
 
+/// Returns the supplied value, and ensures that the value is eagerly loaded into a register.
+#[inline(always)]
+fn ensure_read_ptr<T>(value: *const T) -> *const T {
```

Comment:
Does someone know if there is a `T: FitsIntoARegister` constraint, so I don't have to write the same function three times for `usize`, `*const _` and `*mut _`?